cql3/expr/prepare_expr.cc

```diff
@@ -908,6 +908,17 @@ test_assignment_function_call(const cql3::expr::function_call& fc, data_dictiona
     }
 }
 
+static assignment_testable::test_result expression_test_assignment(const data_type& expr_type,
+                                                                   const column_specification& receiver) {
+    if (receiver.type->underlying_type() == expr_type->underlying_type()) {
+        return assignment_testable::test_result::EXACT_MATCH;
+    } else if (receiver.type->is_value_compatible_with(*expr_type)) {
+        return assignment_testable::test_result::WEAKLY_ASSIGNABLE;
```

Comment:
I think this is correct and consistent with the existing code, but wrong in general. We shouldn't allow assignment from long to timestamp type without a cast. However it will be difficult to deprecate such assignments.

Reply:
AFAIR in another discussion we decided to allow them for now, for compatibility with Cassandra, but eventually we will add a flag that forbids such casts, something like `strict_types: true`.